packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -57,30 +79,22 @@ export class FileNavigatorWidget extends FileTreeWidget {
                         this.model.expandNode(child);
                     }
                 }
-            }),
-            this.navigatorSearch,
-            this.navigatorSearch.onFilteredNodesChanged(nodes => {
-                const node = nodes.find(SelectableTreeNode.is);
-                if (node) {
-                    this.model.selectNode(node);
-                }
-            }),
-            this.searchBox,
-            this.searchBox.onTextChange(data => this.navigatorSearch.filter(data)),
-            this.searchBox.onClose(data => this.navigatorSearch.filter(undefined)),
-            this.searchBox.onNext(() => this.model.selectNextNode()),
-            this.searchBox.onPrevious(() => this.model.selectPrevNode()),
+            })
         ]);
     }
 
-    protected initialize(): void {
-        this.model.onSelectionChanged(selection => {
-            if (this.shell.activeWidget === this) {
-                this.selectionService.selection = selection;
-            }
+    protected updateGlobalSelection(selection: ReadonlyArray<Readonly<SelectableTreeNode>>): void {
+        if (this.shell.activeWidget !== this) {
+            return;
         }
-        );
+        this.selectionService.selection = selection;
+        const removeGlobalSelection = this.toDispose.push(Disposable.create(() =>
+            this.selectionService.selection = undefined
+        ));
+        this.toDispose.push(this.selectionService.onSelectionChanged(() => removeGlobalSelection.dispose()));
```

Comment:
same here

Reply:
where is the caller of `updateGlobalSelection`?